packages/jaeger-ui/src/components/common/ErrorMessage.test.js

```diff
@@ -13,141 +13,124 @@
-      };
-
-      wrapper.setProps({ error });
-      const details = wrapper.find('Details');
-      expect(details.length).toBe(1);
-      const detailsWrapper = details.shallow();
-      expect(detailsWrapper).toMatchSnapshot();
+  it('renders HTTP related data from the error', () => {
+    const error = {
+      message: 'some-http-ish-message',
+      httpStatus: 'value-httpStatus',
+      httpStatusText: 'value-httpStatusText',
+      httpUrl: 'value-httpUrl',
+      httpQuery: 'value-httpQuery',
+      httpBody: 'value-httpBody',
+    };
+
+    render(<ErrorMessage error={error} />);
+    Object.keys(error).forEach(key => {
+      expect(screen.getByText(error[key])).toBeInTheDocument();
     });
+  });
+
+  it('renders truncated body excerpt on large body', () => {
+    const error = {
+      message: 'some-http-ish-message',
+      httpStatus: 'value-httpStatus',
+      httpStatusText: 'value-httpStatusText',
+      httpUrl: 'value-httpUrl',
+      httpQuery: 'value-httpQuery',
+      httpBody: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ac maximus elit. Curabitur non urna in odio dictum porttitor. Fusce sed mi mauris. Sed vehicula mi nec nulla ultricies, a faucibus nisi feugiat. Aliquam varius diam in porttitor maximus. Quisque in varius neque, vel consequat enim. Donec faucibus lorem tortor, a aliquam augue vulputate eu. Pellentesque tincidunt, nisl vitae tristique fringilla, elit augue eleifend leo, eget laoreet turpis justo et dui. Suspendisse nec lacinia tortor, non fermentum arcu. Morbi at nunc nisi. Vestibulum condimentum sollicitudin nibh ut cursus. Nullam neque erat, eleifend eget libero eget, porta maximus diam. Integer ut est congue, placerat ipsum nec, varius ligula. Integer maximus gravida velit quis commodo. Phasellus posuere a nulla id sodales. Aliquam ultrices purus et iaculis imperdiet. Morbi in felis maximus, dictum sapien in, malesuada augue. Duis sit amet tortor ac ante pellentesque iaculis eget vestibulum ex. Nunc malesuada egestas mauris, ut tempus eros hendrerit dui.`,
+    };
+
+    render(<ErrorMessage error={error} />);
 
-    it('renders on missing httpStatus from the error', () => {
-      error = {
-        message: 'some-http-ish-message',
-        httpStatusText: 'value-httpStatusText',
-        httpUrl: 'value-httpUrl',
-        httpQuery: 'value-httpQuery',
-        httpBody: 'value-httpBody',
-      };
-      wrapper.setProps({ error });
-      const details = wrapper.find('Details');
-      expect(details.length).toBe(1);
-      const detailsWrapper = details.shallow();
-      expect(detailsWrapper).toMatchSnapshot();
+    // All the keys expect httpBody should be present as is
+    Object.keys(error).forEach(key => {
+      if (key !== 'httpBody') expect(screen.getByText(error[key])).toBeInTheDocument();
     });
+
+    // The body should be truncated
+    expect(screen.getByText(`${error.httpBody.slice(0, 1021).trim()}...`)).toBeInTheDocument();
```

Comment:
what us this magic number 1021? If the component has a specific limit, let's expose that as a constant from the component and use here.

Reply:
It's the maximum limit `1024` - length of `...`. Will expose it as a constant.